### PR TITLE
Fix column_name and null generation

### PR DESCRIPTION
**Pull request by ksneab7**

### CODE REVIEW

**Review by taylorfturner — APPROVED** [review]

**Review by taylorfturner — CHANGES_REQUESTED** [review]
let's validate tests

File: tests/test_generators.py [hunk 1]
```diff
@@ -384,7 +405,13 @@ def test_get_ordered_column_integration(self, mock_report, mock_warning):
...
-        categories = ["int", "string", "categorical", "float", "datetime"]
+        categories = [
+            "test_column_1",
+            "test_column_2",
+            "test_column_3",
+            "test_column_4",
+            "test_column_5",
+        ]
 
         expected_data = [dict(zip(categories, item)) for item in expected_array]
```
Comment (taylorfturner) [inline tests/test_generators.py:416]:
```suggestion
        expected_column_names = [
            "test_column_1",
            "test_column_2",
            "test_column_3",
            "test_column_4",
            "test_column_5",
        ]

        expected_data = [dict(zip(expected_column_names, item)) for item in expected_array]
```

**Review by taylorfturner — CHANGES_REQUESTED** [review]

**Review by taylorfturner — DISMISSED** [review]

**Review by taylorfturner — APPROVED** [review]

File: synthetic_data/distinct_generators/null_generator.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
+"""Contains a random Null generator."""
```
Comment (ksneab7) [inline synthetic_data/distinct_generators/null_generator.py:1]: This function is used to generate fully null columns

File: synthetic_data/generators.py [hunk 3]
```diff
@@ -105,49 +107,46 @@ def _generate_uncorrelated_column_data(self, num_samples):
...
-            if not generator_name:
-                logging.warning(
-                    f"Generator of type {generator_name} is not implemented."
-                )
-                continue
+            column_header = col_.get("column_name", None)
 
             col_["rng"] = self.rng
             col_["num_rows"] = num_samples
+            if generator_name:
```
Comment (ksneab7) [inline synthetic_data/generators.py:114]: This is needed in place of the removed warning, as we account for a fully nulled column

File: synthetic_data/generators.py [hunk 4]
```diff
@@ -105,49 +107,46 @@ def _generate_uncorrelated_column_data(self, num_samples):
...
-            if generator_name == "float":
-                col_["precision"] = int(
-                    col_["statistics"].get("precision", None).get("max", None)
-                )
+                if generator_name == "float":
+                    col_["precision"] = int(
+                        col_["statistics"].get("precision", None).get("max", None)
+                    )
+            else:
+                generator_name = "null_generator"
```
Comment (taylorfturner) [inline synthetic_data/generators.py:149]: might do an elif here for readability

File: synthetic_data/generators.py [hunk 5]
```diff
@@ -166,7 +167,9 @@ def _generate_uncorrelated_column_data(self, num_samples):
                     )
                 )
             else:
-                if col_["order"] is not None:
+                if (not generator_name == "null_generator") and col_[
+                    "order"
+                ] is not None:
```
Comment (taylorfturner) [inline synthetic_data/generators.py:172]: just an else here (only way it gets is if `generator_name` not in the dictionary keys) and just `logging.warning`)

**Review by taylorfturner — CHANGES_REQUESTED** [review]

File: synthetic_data/generators.py (hunk 5 above)
Comment (ksneab7) [reply synthetic_data/generators.py:172]: The warning above is not needed anymore just because of the null columns now being accounted for.

**Review by danielbarcklow — APPROVED** [review]
LGTM

**Review by taylorfturner — APPROVED** [review]
LGTM

**Pull request merged**